services/src/grpc-speaker/grpc-service/src/main/java/org/openkilda/grpc/speaker/client/GrpcSession.java

```diff
@@ -0,0 +1,376 @@
+                .setUsername(user)
+                .setPassword(pass)
+                .build();
+
+        log.debug("Performs auth user request to switch {} with user {}", address, user);
+
+        GrpcResponseObserver<CliReply> observer = new GrpcResponseObserver<>();
+
+        stub.setLoginDetails(authUser, observer);
+        return observer.future;
+    }
+
+    /**
+     * Performs switch status request.
+     *
+     * @return {@link CompletableFuture} with operation result.
+     */
+    public CompletableFuture<Optional<StatusSwitch>> showSwitchStatus() {
+        GrpcResponseObserver<StatusSwitch> observer = new GrpcResponseObserver<>();
+
+        log.debug("Getting switch status for switch {}", address);
+
+        stub.showStatusSwitch(StatusSwitch.newBuilder().build(), observer);
+        return observer.future
+                .thenApply(responses -> responses.stream().findFirst());
+    }
+
+    /**
+     * Performs set logical port config switch request.
+     *
+     * @return {@link CompletableFuture} with operation result.
+     */
+    public CompletableFuture<List<CliReply>> setLogicalPort(LogicalPortDto port) {
+        Objects.requireNonNull(port.getLogicalPortNumber(), "Logical port number must not be null");
+        Objects.requireNonNull(port.getPortNumbers(), "Port number must not be null");
+
+        LogicalPort request = LogicalPort.newBuilder()
+                .addAllPortno(port.getPortNumbers())
+                .setLogicalportno(port.getLogicalPortNumber())
+                .build();
+
+        log.debug("About to create logical port: {}", request);
+
+        GrpcResponseObserver<CliReply> observer = new GrpcResponseObserver<>();
+
+        stub.setConfigLogicalPort(request, observer);
+        return observer.future;
+    }
+
+    /**
+     * Performs show switch logical port request.
+     *
+     * @param port the {@link LogicalPortDto} instance.
+     * @return {@link CompletableFuture} with operation result.
+     */
+    public CompletableFuture<Optional<LogicalPort>> showConfigLogicalPort(Integer port) {
+        Objects.requireNonNull(port, "Port must not be null");
+        LogicalPort request = LogicalPort.newBuilder()
+                .setLogicalportno(port)
+                .build();
+
+        log.debug("Reading logical port {} from the switch: {}", port, address);
+
+        GrpcResponseObserver<LogicalPort> observer = new GrpcResponseObserver<>();
+        stub.showConfigLogicalPort(request, observer);
+
+        return observer.future
+                .thenApply(responses -> responses.stream().findFirst());
+    }
+
+    /**
+     * Performs dump switch logical port request.
+     *
+     * @return {@link CompletableFuture} with operation result.
+     */
+    public CompletableFuture<List<LogicalPort>> dumpLogicalPorts() {
+        LogicalPort request = LogicalPort.newBuilder().build();
+
+        log.debug("Getting all logical ports for switch: {}", address);
```

Comment:
I'm not sure whether it should be debug or info tbh(in all methods within the class)

Reply:
replaced with INFO level